### PR TITLE
ci - drop extra CPU tests in CUDA job

### PR DESCRIPTION
**Pull request by jeremylt**
Small oops

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -196,7 +196,7 @@ noether-cuda:
 # -- PETSc with CUDA (minimal)
     - export PETSC_DIR=/projects/petsc PETSC_ARCH=mpich-cuda-O PETSC_OPTIONS='-use_gpu_aware_mpi 0' && git -C $PETSC_DIR -c safe.directory=$PETSC_DIR describe
     - echo "-------------- PETSc ---------------" && make -C $PETSC_DIR info
-    - make -k -j$((NPROC_GPU / NPROC_POOL)) JUNIT_BATCH="cuda" junit BACKENDS="$BACKENDS_GPU" search="petsc fluids-navierstokes solids"
+    - make -k -j$((NPROC_GPU / NPROC_POOL)) BACKENDS="$BACKENDS_GPU" JUNIT_BATCH="cuda" junit search="petsc fluids-navierstokes solids"
```
Comment (jrwrigh) [inline .gitlab-ci.yml:199]: Order of pass options/variables shouldn't change how make works... right?

Comment (jrwrigh) [reply .gitlab-ci.yml:199]: I don't know how to double check `junit`, but 
```console
make BACKENDS='/cpu/self/memcheck/blocked' NPROC_POOL=3 test search='fluids-n' -j
```

and

```console
make NPROC_POOL=3 test BACKENDS='/cpu/self/memcheck/blocked' search='fluids-n' -j
```

Both run with `/cpu/self/memcheck/blocked`

Comment (jeremylt) [reply .gitlab-ci.yml:199]: I agree it shouldn't matter, but now more tests are skipped on Noether so :shrug:

Comment (jrwrigh) [reply .gitlab-ci.yml:199]: How are you seeing that?

Comment (jeremylt) [reply .gitlab-ci.yml:199]: You can see the extra CPU runs here: https://gitlab.com/libceed/libCEED/-/pipelines/1352879258/test_report?job_name=noether-cuda

but now I'm still seeing them for this branch too?

Comment (jrwrigh) [reply .gitlab-ci.yml:199]: Huh. Yeah, I see them now. Not sure why they're happening though.

Comment (jeremylt) [reply .gitlab-ci.yml:203]: ooh, I think I might have actually figured it out

Comment (jrwrigh) [reply .gitlab-ci.yml:203]: Can `make clean` be parallelized?

Comment (jeremylt) [reply .gitlab-ci.yml:203]: got it!

Comment (jrwrigh) [reply .gitlab-ci.yml:203]: Just for my own sanity, the change in argument order isn't necessary anymore, right? It's just the `make clean` that was causing the issue?

**Review by jrwrigh — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (jeremylt) [reply .gitlab-ci.yml:203]: Yup. I left it just because its odd to have one single command out of order

**Pull request merged**